website/content/specification/syntax/constraints.md

```diff
@@ -164,6 +164,30 @@ And the following document.
 The expect constraint would pass for each `sibling` in the `parent` named "p1", and would fail for each `sibling` in the `parent` named "p2".
 
 
+## `index` constraints
+
+The `index` constraint is a type of Metaschema constraint that defines an index of document instance nodes addressable by key.
+
+The `@name` attribute of an `<index>` constraint specifies the identity of the index. The constraint MUST define the name.
+
+The `@target` attribute of an `<index>` constraint defines the node(s) in a document instance to index. The index MUST define a [`@target`](#target) with a Metapath expression. The processor MUST index only The document instance node(s) resulting from its evaluation.
+
+The `<key-field/>` child element of an `<index>` constraint defines the attribute or element value that is the key for each entry in the index. The `<key-field/>` element MUST define a [`@target`](#target) attribute with a Metapath expression evaluated relative to [the evaluation focus](#constraint-processing) of each entry of the matches of the constraint's `@target`.
```

Comment:
This should use Metaschema terminology instead of XML terminology. e.g. flag vs attribute.

Reply:
I thought we were really describing the XML syntax of the domain specific language for Metaschema definitions and it might get confusing. I definitely adjust it.

Reply:
Fixed in https://github.com/usnistgov/metaschema/pull/494/commits/933b6455cd014737a3d7d9ae4df4e41cd65a9666.